js.go

```diff
@@ -150,6 +150,14 @@ func ApiPrefix(pre string) JSOpt {
 		if !strings.HasSuffix(js.pre, ".") {
 			js.pre = js.pre + "."
 		}
+
+		for _, ch := range js.pre {
```

Comment:
Need to tokenize first. So do `strings.Split(js.pre, ".")` and check against the tokens. Although not common this is technically legal..

`foo.*>*'

Reply:
Ah, okay. Just pushed some more tests and changed the validation logic a bit.

Added these cases:
```js
{
	prefix:     "foo",
	wantPrefix: "foo.",
},
{
	prefix:     "foo.",
	wantPrefix: "foo.",
},
{
	prefix:     "foo.>",
	wantPrefix: "foo.>",
},
{
	prefix:  "foo.*",
	wantErr: true,
},
{
	prefix:  "foo.>.bar",
	wantErr: true,
},
```
https://github.com/nats-io/nats.go/pull/617/commits/b2987b5e47db95f4631943d12446360d9f2271be#diff-6277b08b1b6714cf4cd8c16c727011ff2c99d3a8d1b41a0d3fc2d81a49b1f3f5R2609